tests/ui/unnecessary_to_owned.fixed

```diff
@@ -426,3 +426,32 @@ mod issue_9504 {
         foo(std::path::PathBuf::new().to_string_lossy().to_string()).await;
     }
 }
+
+mod issue_9771a {
+    #![allow(dead_code)]
+
+    use std::marker::PhantomData;
+
+    pub struct Key<K: AsRef<[u8]>, V: ?Sized>(K, PhantomData<V>);
+
+    impl<K: AsRef<[u8]>, V: ?Sized> Key<K, V> {
+        pub fn new(key: K) -> Key<K, V> {
+            Key(key, PhantomData)
+        }
+    }
+
+    pub fn pkh(pkh: &[u8]) -> Key<Vec<u8>, String> {
+        Key::new([b"pkh-", pkh].concat().to_vec())
+    }
+}
+
+mod issue_9771b {
+    #![allow(dead_code)]
+
+    pub struct Key<K: AsRef<[u8]>>(K);
+
+    pub fn from(c: &[u8]) -> Key<Vec<u8>> {
+        let v = [c].concat();
+        Key(v.to_vec())
```

Comment:
Note that this issue was not a false positive, ideally this should be corrected to `Key(v)`, as `.concat()` already returns a `Vec<u8>`.
```suggestion
        Key(v)
```

Reply:
I take your point, but I think we have to treat the issue as a false positive for now.

Replacing `v.to_vec()` with `v` would move `v`, and `unnecessary_to_owned` doesn't currently have a way to tell whether a value can be moved.

I suspect it would be possible to incorporate something like `possible_borrowers` from `redundant_clone`. But I also think that would be a significant change and, hence, an enhancement.

Reply:
Alright, just wanted to make sure that you understood the issue correctly. I don't know much about the internals of clippy, so I can't help you with implementing this.